Quick Notifications

CxRes opened the pull request:
Solid Quick Notifications is a proposal to use `Accept-Events` and `Events` header fields that I invented in [PREP](https://cxres.github.io/prep/draft-gupta-httpbis-per-resource-events.html) to allow Clients to negotiate subscriptions i.e. request a notification channel, using a mechanism identical to content negotiation in HTTP. With this one can request a notification channel in a single request, and given that a client will anyway start with a GET request to the topic, the subscription step is effectively free when using proactive negotiation. While not as efficient as PREP over a single topic, this would outperform it when requesting notifications for multiple topic resources.

For public channels, I am proposing to introduce a `Notification-Channels` header field to, well, advertise the channels. 

I believe this proposal solves all the issues raised by @TimBL in #110 as well as the concern he raised in the [notification panel meeting on 12-01-2023](https://github.com/solid/notifications-panel/blob/main/meetings/2023-01-12.md) while remaining completely backwards compatible (apart from an extremely minor caveat) with Solid Notifications. One can even use it alongside the existing specification, providing clients a choice of mechanisms for discovery and subscription of notifications.

**Updated** [HTML Preview](https://htmlpreview.github.io/?https://raw.githubusercontent.com/solid/notifications/ed3b8292a4c3652e9b0aa7fa488f451f70f1ac16/quick-protocol.html) | [Diff with main branch](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fraw.githubusercontent.com%2Fsolid%2Fnotifications%2F837221ddeeeb69ac123f1a766765ddf8f455f7e9%2Fprotocol.html&doc2=https%3A%2F%2Fraw.githubusercontent.com%2Fsolid%2Fnotifications%2Fed3b8292a4c3652e9b0aa7fa488f451f70f1ac16%2Fquick-protocol.html) | [Diff with backported changes to SNP](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fraw.githubusercontent.com%2Fsolid%2Fnotifications%2Fe1a7800581a6892102fc0428a653e55a21992afc%2Fprotocol.html&doc2=https%3A%2F%2Fraw.githubusercontent.com%2Fsolid%2Fnotifications%2Fed3b8292a4c3652e9b0aa7fa488f451f70f1ac16%2Fquick-protocol.html)

CxRes committed:
Add Quick Notifications

Adds the Solid Quick Notifications Protocol that enables Solid Servers to directly provide clients with notification channels and/or let them negotiate subscriptions using HTTP headers within any GET requests.

csarven reviewed (commented):
Just a Quick Comment for now. Will leave a separate thorough review later once I understand the technicals. High level:

If possible, I'd much rather see how this work can either get incorporated into the current Solid Notifications Protocol  or made possible to only extend it. If Solid Notifications Protocol doesn't make it clear how extension could occur, we can improve on that. Essentially, this kind of work should stand on its own as opposed to a "derived from" type of thing.

CxRes commented:
@csarven This is to start a conversation. My long term suggestion (especially if such a work can get funded), is to reorganize Solid Notifications in a number of specifications:
+ Solid Notifications: Discovery and Subscription
+ Solid Notifications: Quick Discovery and Notifications
+ Solid Notifications: Channels
+ Solid Notifications: <Channel Type>
+ Solid Notifications: Messages for RDF resources
+ Solid Notifications: Messages for <content-type or content category>

Really, the bulk of work has to be done in the last two! It also makes it easier for someone entering Solid Notifications to work with more palatable chunks! This might be a good opportunity to rebrand as well, Since these notifications can easily be used outside Solid.

elf-pavlik commented on README.md:
```diff
@@ -12,6 +12,12 @@ These reports are incubated by the [Solid Notifications Panel](https://github.co
   * [Solid Notifications Protocol](https://solid.github.io/notifications/protocol)
 
+### Quick Protocol
+* Latest Published Version:
+  * [Solid Quick Notifications Protocol](https://solidproject.org/TR/quick-notifications-protocol)
```
404 I would suggest to add that link once it gets published

CxRes replied:
My intention was that when the PR gets accepted, it will be created.

Happy to comment it out if it is a problem...

elf-pavlik commented on quick-protocol.html:
Since this is an edit to the current notifications protocol document, I will need to go through the [diff](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fsolid.github.io%2Fnotifications%2Fprotocol&doc2=https%3A%2F%2Fraw.githubusercontent.com%2Fsolid%2Fnotifications%2F56ddf9f68c38c59a5ad4affe9a8fc919882882b5%2Fquick-protocol.html)

CxRes replied:
Use the diff I have created in the first comment. That compares with the latest SNP version!

csarven commented:
Can we take this up in an STM? I'd like to understand better whether this should be incorporated into SNP or being proposed as a new work item or what...

.. In addition to noting any early commitments for implementation.

CxRes commented:
First and foremost, I would like for the reviewers to verify if my proposal is functionally correct!

We can decide then how to proceed, an extension, an independent work item or re-organization of notification specs. I think all three are viable, but it is really a question of resources and participation.

I would like an STM as well, perhaps after a first round of reviews!

TallTed commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+                <div datatype="rdf:HTML" property="schema:description">
+                  <p><em>This section is non-normative.</em></p>
+
+                  <p>The Solid Quick Notification protocol defines a mechanism for client applications to discover notification services available for any given resource.</p>
```
```suggestion
                  <p>The Solid Quick Notification protocol defines a mechanism whereby client applications can discover notification services available for any given resource.</p>
```

TallTed commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+                    <figcaption property="schema:name">Solid Quick Notifications Flow Discovery</figcaption>
+                  </figure>
+
+                  <p>A Resource Server allows Subscription Clients to request a customised Notification Channel.</p>
```
```suggestion
                  <p>A Resource Server allows Subscription Clients to request customised Notification Channels.</p>
```

TallTed commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+
+                  <p>A notification channel allows a Notification Sender to send notifications to a Notifications Receiver. A notification channel builds upon one of many established web protocols to exchange data, determining its <em>type</em>.</p>
+
+                  <p>The following diagram illustrates the flow of data for those notification channel types in which a client establishes a subscription and then maintains a persistent connection to a notifications source are illustrated by the following diagram.</p>
```
```suggestion
                  <p>The following diagram illustrates the flow of data for those notification channel types in which a client establishes a subscription and then maintains a persistent connection to a notifications source.</p>
```

TallTed commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+                  <figure id="solid-quick-notifications-flow-receivefrom" rel="schema:hasPart" resource="#solid-quick-notifications-flow-receivefrom">
+                    <img rel="schema:image" src="notifications-flow-receivefrom.mmd.svg" width="800" />
+
+                    <figcaption property="schema:name">Solid Quick Notifications Flow Receive From, where the receiver establishes connection to sender.</figcaption>
```
```suggestion
                    <figcaption property="schema:name">Solid Quick Notifications Flow Receive From, where the receiver establishes a connection to a sender.</figcaption>
```

TallTed commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+                  <figure id="solid-quick-notifications-flow-sendto" rel="schema:hasPart" resource="#solid-quick-notifications-flow-sendto">
+                    <img rel="schema:image" src="notifications-flow-sendto.mmd.svg" width="800" />
+
+                    <figcaption property="schema:name">Solid Quick Notifications Flow Send To, where the sender delivers notifications to receiver.</figcaption>
```
```suggestion
                    <figcaption property="schema:name">Solid Quick Notifications Flow Send To, where the sender delivers notifications to a receiver.</figcaption>
```

TallTed commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+                  <section id="specification-category" inlist="" rel="schema:hasPart" resource="#specification-category" typeof="skos:ConceptScheme">
+                    <h4 property="schema:name skos:prefLabel">Specification Category</h4>
+                    <div datatype="rdf:HTML" property="schema:description">
+                      <p property="skos:definition">The <span about="" rel="spec:specificationCategory" resource="#specification-category">Solid Quick Notifications Protocol</span> identifies the following <cite><a href="https://www.w3.org/TR/spec-variability/#spec-cat" rel="dcterms:subject" resource="spec:SpecificationCategory">Specification Category</a></cite> to distinguish the types of confor ...
```
```suggestion
                      <p property="skos:definition">The <span about="" rel="spec:specificationCategory" resource="#specification-category">Solid Quick Notifications Protocol</span> identifies the following <cite><a href="https://www.w3.org/TR/spec-variability/#spec-cat" rel="dcterms:subject" resource="spec:SpecificationCategory">Specification Categories</a></cite> to distinguish the types of conformance: <span rel="skos:hasTopConcept" resource="spec:API">API</span>, <span rel="skos:hasTopConcept" resource="spec:NotationSyntax">notation/syntax</span>, <span rel="skos:hasTopConcept" resource="spec:SetOfEvents">set of events</span>, <span rel="skos:hasTopConcept" resource="spec:Protocol">protocol</span>.</p>
```

TallTed commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+                        <dt about="#ResourceServer" property="skos:prefLabel" rel="skos:broadMatch" resource="spec:RespondingAgent" typeof="skos:Concept"><dfn id="ResourceServer">Resource Server</dfn></dt>
+                        <dd about="#ResourceServer" property="skos:definition">A <em>resource server</em> builds on an HTTP <cite>server</cite> [<cite><a class="bibref" href="#bib-rfc9110">RFC9110</a></cite>] by defining media types, HTTP header fields, and the behaviour of resources, as identified by link relations.</dd>
+                        <dt about="#SubscriptionClient" property="skos:prefLabel" rel="skos:broadMatch" resource="spec:Consumer" typeof="skos:Concept"><dfn id="SubscriptionClient">Subscription Client</dfn></dt>
+                        <dd about="#SubscriptionClient" property="skos:definition">A <em>Subscription Client</em> that builds on HTTP <cite>client</cite> [<cite><a class="bibref" href="#bib-rfc9110">RFC9110</a></cite>] and [<cite><a class="bibref" href="#bib-fetch">FETCH</a></cite>] by defining behaviour in terms of fetching across the platform.</dd>
```
```suggestion
                        <dd about="#SubscriptionClient" property="skos:definition">A <em>Subscription Client</em> builds on an HTTP <cite>client</cite> [<cite><a class="bibref" href="#bib-rfc9110">RFC9110</a></cite>] and [<cite><a class="bibref" href="#bib-fetch">FETCH</a></cite>] by defining behaviour in terms of fetching across the platform.</dd>
```

CxRes replied:
Sarven's language. (Reminder to self!) To be made consistent in both documents.

CxRes replied:
@csarven, Some products say "builds on" and other "that builds on". Please pick one for consistency!

TallTed commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+                        <dt about="#SubscriptionClient" property="skos:prefLabel" rel="skos:broadMatch" resource="spec:Consumer" typeof="skos:Concept"><dfn id="SubscriptionClient">Subscription Client</dfn></dt>
+                        <dd about="#SubscriptionClient" property="skos:definition">A <em>Subscription Client</em> that builds on HTTP <cite>client</cite> [<cite><a class="bibref" href="#bib-rfc9110">RFC9110</a></cite>] and [<cite><a class="bibref" href="#bib-fetch">FETCH</a></cite>] by defining behaviour in terms of fetching across the platform.</dd>
+                        <dt about="#NotificationSender" property="skos:prefLabel" rel="skos:broadMatch" resource="spec:RespondingAgent" typeof="skos:Concept"><dfn id="NotificationSender">Notification Sender</dfn></dt>
+                        <dd about="#NotificationSender" property="skos:definition" rel="skos:broadMatch" resource="spec:ProducerOfContent">A <em>Notification Sender</em> is the sender of a notification acting in conformance to the product class of a notification channel type.</dd>
+                        <dt about="#NotificationReceiver" property="skos:prefLabel" rel="skos:broadMatch" resource="spec:RespondingAgent" typeof="skos:Concept"><dfn id="NotificationReceiver">Notification Receiver</dfn></dt>
+                        <dd about="#NotificationReceiver" property="skos:definition" rel="skos:broadMatch" resource="spec:Consumer">A <em>Notification Receiver</em> is the recipient of a notification acting in conformance to the product class of a notification channel type.</dd>
```
```suggestion
                        <dd about="#NotificationSender" property="skos:definition" rel="skos:broadMatch" resource="spec:ProducerOfContent">A <em>Notification Sender</em> is the sender of a notification acting in conformance with the product class of a notification channel type.</dd>
                        <dt about="#NotificationReceiver" property="skos:prefLabel" rel="skos:broadMatch" resource="spec:RespondingAgent" typeof="skos:Concept"><dfn id="NotificationReceiver">Notification Receiver</dfn></dt>
                        <dd about="#NotificationReceiver" property="skos:definition" rel="skos:broadMatch" resource="spec:Consumer">A <em>Notification Receiver</em> is the recipient of a notification acting in conformance with the product class of a notification channel type.</dd>
```

TallTed commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+                  <section id="interoperability" inlist="" rel="schema:hasPart" resource="#interoperability" typeof="skos:ConceptScheme">
+                    <h4 property="schema:name skos:prefLabel">Interoperability</h4>
+                    <div datatype="rdf:HTML" property="schema:description">
+                      <p property="skos:definition">Interoperability occurs between <a href="#subscription-client-resource-server-interoperability" rel="skos:hasTopConcept">Subscription Client–Resource Server</a> or <a href="#notification-sender-notification-receiver-interoperability" rel="skos:hasTopConcept">Notification Sender–Notification Receiver</a>, as defined by the <a href="" rel="foaf:pa ...
```
```suggestion
                      <p property="skos:definition">Interoperability occurs between <a href="#subscription-client-resource-server-interoperability" rel="skos:hasTopConcept">Subscription Client↔Resource Server</a> or <a href="#notification-sender-notification-receiver-interoperability" rel="skos:hasTopConcept">Notification Sender↔Notification Receiver</a>, as defined by the <a href="" rel="foaf:page">Solid Quick Notifications Protocol</a>.</p>
```

CxRes replied:
Need to check with Sarven on the arrows.

TallTed commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+                      <p property="skos:definition">Interoperability occurs between <a href="#subscription-client-resource-server-interoperability" rel="skos:hasTopConcept">Subscription Client–Resource Server</a> or <a href="#notification-sender-notification-receiver-interoperability" rel="skos:hasTopConcept">Notification Sender–Notification Receiver</a>, as defined by the <a href="" rel="foaf:pa ...
+
+                      <dl>
+                        <dt about="#subscription-client-resource-server-interoperability" property="skos:prefLabel" typeof="skos:Concept"><dfn id="subscription-client-resource-server-interoperability">Subscription Client–Resource Server interoperability</dfn></dt>
+                        <dd about="#subscription-client-resource-server-interoperability" property="skos:definition">Interoperability of implementations for <a href="#SubscriptionClient" rel="rdfs:seeAlso">Subscription Clients</a> and <a href="#ResourceServer" rel="rdfs:seeAlso">Resource Servers</a> is tested by evaluating an implementation’s ability to request and respond to HTTP messages that c ...
+                        <dt about="#notification-sender-notification-receiver-interoperability" property="skos:prefLabel" typeof="skos:Concept"><dfn id="notification-sender-notification-receiver-interoperability">Notification Sender–Notification Receiver interoperability</dfn></dt>
+                        <dd about="#notification-sender-notification-receiver-interoperability" property="skos:definition">Interoperability of implementations for <a href="#NotificationSender" rel="rdfs:seeAlso">Notification Sender</a> and <a href="#NotificationReceiver" rel="rdfs:seeAlso">Notification Receiver</a> is tested by evaluating an implementation’s ability to produce and consume content ...
```
```suggestion
                        <dt about="#subscription-client-resource-server-interoperability" property="skos:prefLabel" typeof="skos:Concept"><dfn id="subscription-client-resource-server-interoperability">Subscription Client↔Resource Server interoperability</dfn></dt>
                        <dd about="#subscription-client-resource-server-interoperability" property="skos:definition">Interoperability of <a href="#SubscriptionClient" rel="rdfs:seeAlso">Subscription Client</a> and <a href="#ResourceServer" rel="rdfs:seeAlso">Resource Server</a> implementations is tested by evaluating an implementation’s ability to request and respond to HTTP messages that conform to the Solid Quick Notifications Protocol.</dd>
                        <dt about="#notification-sender-notification-receiver-interoperability" property="skos:prefLabel" typeof="skos:Concept"><dfn id="notification-sender-notification-receiver-interoperability">Notification Sender↔Notification Receiver interoperability</dfn></dt>
                        <dd about="#notification-sender-notification-receiver-interoperability" property="skos:definition">Interoperability of <a href="#NotificationSender" rel="rdfs:seeAlso">Notification Sender</a> and <a href="#NotificationReceiver" rel="rdfs:seeAlso">Notification Receiver</a> implementations is tested by evaluating an implementation’s ability to produce and consume content that adheres to the <a href="#notification-message-data-model" rel="rdfs:seeAlso">notification message data model</a> and conforms to a particular <a href="#notification-channel-types" rel="rdfs:seeAlso">notification channel type</a>.</dd>
```

CxRes replied:
Again, for @csarven to review!

TallTed commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+          <section id="protocol" inlist="" rel="schema:hasPart" resource="#protocol">
+            <h2 property="schema:name">Protocol</h2>
+            <div datatype="rdf:HTML" property="schema:description">
+              <p>The Solid Quick Notifications Protocol specifies a mechanism for discovery of notification capabilities on resources, customisation of notification channels through subscriptions, capabilities of notification channels, and methods for establishing a connection to a notification channel.</p>
```
```suggestion
              <p>The Solid Quick Notifications Protocol specifies mechanisms for discovery of notification capabilities on resources, customisation of notification channels through subscriptions, capabilities of notification channels, and methods for establishing a connection to a notification channel.</p>
```

TallTed commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+                    <div>
+                      <p><a href="#ResourceServer">Resource Servers</a> can advertise their ability to create <a href="#notification-channel">notification channels</a> to provide notifications:</p>
+                        <ul>
+                          <li>for protected resource, which relies on capability URL to protect the channel;</li>
```
```suggestion
                          <li>for a protected resource, which relies on a capability URL to protect the channel</li>
```

TallTed reviewed (changes requested):
It's a start...

CxRes commented:
@TallTed A bunch of these changes relate to the original text. I will propose another PR to the main branch to fix it there.

CxRes commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+                  <section id="subscription-request" inlist="" rel="schema:hasPart" resource="#subscription-request">
+                    <h4 property="schema:name">Subscription Request</h4>
+
+                    <p about="#subscription-request" property="skos:definition" typeof="skos:Concept">A <strong><dfn property="skos:prefLabel">subscription request</dfn></strong> is an HTTP request from a <a href="#SubscriptionClient">Subscription Client</a> that additionally requests a <a href="#ResourceServer">Resource Server</a> to create a desired <a href="#notification-channel">notification  ...
```
```suggestion
                    <p about="#subscription-request" property="skos:definition" typeof="skos:Concept">A <strong><dfn property="skos:prefLabel">subscription request</dfn></strong> is an HTTP request from a <a href="#SubscriptionClient">Subscription Client</a> that additionally requests that the <a href="#ResourceServer">Resource Server</a> create a desired <a href="#notification-channel">notification channel</a> through which it is able to receive notifications about one or more <a href="#topic-resource">topic resources</a>.</p>
```

CxRes commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+
+                    <p about="#subscription-request" property="skos:definition" typeof="skos:Concept">A <strong><dfn property="skos:prefLabel">subscription request</dfn></strong> is an HTTP request from a <a href="#SubscriptionClient">Subscription Client</a> that additionally requests a <a href="#ResourceServer">Resource Server</a> to create a desired <a href="#notification-channel">notification  ...
+
+                    <p about="" id="subscription-client-subscription-request-header" rel="spec:requirement" resource="#subscription-client-subscription-request-header"><span property="spec:statement"><a rel="spec:requirementSubject" href="#SubscriptionClient">Subscription Clients</a> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> use the <code>Accept-Events</code> header field ...
+
+                    <p about="" id="subscription-client-subscription-request-protocol-parameters" rel="spec:requirement" resource="#subscription-client-subscription-request-protocol-parameters"><span property="spec:statement"><a rel="spec:requirementSubject" href="#SubscriptionClient">Subscription Clients</a> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> set the parameters of ...
+
+                  </section>
+
+                  <section id="subscription-response" inlist="" rel="schema:hasPart" resource="#subscription-reponse">
+                    <h4 property="schema:name">Subscription Response</h4>
+
+                    <p about="#subscription-response" property="skos:definition" typeof="skos:Concept">A <strong><dfn property="skos:prefLabel">subscription response</dfn></strong> is an HTTP response to a <a href="#subscription-request">subscription request</a> by a <a href="#ResourceServer">Resource Server</a> that either describes the created <a href="#notification-channel">notification channe ...
+
+                    <p about="" id="resource-server-subscription-reponse-header" rel="spec:requirement" resource="#resource-server-subscription-reponse-header"><span property="spec:statement"><a rel="spec:requirementSubject" href="#ResourceServer">Resource Servers</a> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> use the <code>Events</code> header field [<cite><a class="bibre ...
```
"solid" should be lowercase in the headers. (Reminder to self) fix after the review!

CxRes replied:
Fixed!

CxRes committed:
Apply suggestions from TallTed's review

Fixes multiple typo and language errors.

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

CxRes committed:
TallTed's Review Fixes Backported

Fixes typos and language in SNP, arising from TallTed's review of SQNP.

CxRes commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+                        <p><a href="#ResourceServer">Resource Servers</a> that want to enable <a href="#SubscriptionClient">Subscription Clients</a> to discover existing <a href="#notification-channel">notification channels</a> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> include, in response to <code>HEAD</code> or a <code>GET</code> request,  <code>Notification-Channels</c ...
+
+                        <ul>
+                          <li><code>Solid</code> as a List Item [<cite><a class="bibref" href="#bib-sf">STRUCTURED-FIELDS</a></cite>], and</li>
+                          <li>parameters of <a href="#notification-channel">notification channel</a> set according to the <a href="#notification-channel-data-model" rel="rdfs:seeAlso">Notification Channel Data Model</a>.</li>
```
Will revisit `<li>` punctuation later. My preferences are clearly inconsistent with the document!

CxRes commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+                      <div property="spec:statement">
+                        <p><a href="#ResourceServer">Resource Servers</a> that want to enable <a href="#SubscriptionClient">Subscription Clients</a> to request a <a href="#notification-channel">notification channel</a> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> include, in response to a <code>HEAD</code> or a <code>GET</code> request, the <code>Accept-Events</code> header  ...
+                        <ul>
+                          <li><code>Solid</code> as a List Item [<cite><a class="bibref" href="#bib-sf">STRUCTURED-FIELDS</a></cite>], and</li>
+                          <li>parameters for each <a href="#notification-channel-types" rel="rdfs:seeAlso">notification channel type</a> set according to <a href="#subscription-discovery-data-model" rel="rdfs:seeAlso">Subscription Discovery Data Model</a>.</li>
```
Will revisit <li> punctuation later. My preferences are clearly inconsistent with the document!

CxRes commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+                      <p><a href="#ResourceServer">Resource Servers</a> can advertise their ability to create <a href="#notification-channel">notification channels</a> to provide notifications:</p>
+                        <ul>
+                          <li>for protected resource, which relies on capability URL to protect the channel;</li>
+                          <li>with specific <a href="#features">features</a> requested by the <a href="#SubscriptionClient">Subscription Client</a>.</li>
```
Will revisit <li> punctuation later. My preferences are clearly inconsistent with the document!

CxRes commented:
@TallTed Thanks for detailed review. I have accepted (and backported) most of your suggestions. The only changes that I have not accepted have to do with `<li>` punctuation, which I will revisit later.

@csarven Can you please look at Conformance section suggestions made by Ted (since these are really fixes to SNP). If you agree, I will merge them in and backport them to `protocol.html`.

CxRes committed:
On header reference to solid must be lowercase

Fixed inconsistent use of `solid` as a protocol identifier on `Accept-Events` and `Events` header field to lowercase.

CxRes committed:
Typo Fixed

Fixed a spurious text on line 671

CxRes commented on quick-protocol.html:
```diff
@@ -0,0 +1,1186 @@
+              <section id="subscription" inlist="" rel="schema:hasPart" resource="#subscription">
+                <h3 property="schema:name">Subscription</h3>
+                <div datatype="rdf:HTML" property="schema:description">
+                  <p>The subscription mechanism enables <a href="#SubscriptionClient">Subscription Clients</a> to request a <a href="#ResourceServer">Resource Server</a> to create a customized notification channel for receiving notifications for changes on a particular topic resource.</p>
```
Change was accepted, but with a slightly modified language!

CxRes committed:
Fix Examples

Fixed example code based on discussion with elf Pavlik in the Solid STM on 5 March 2024.
Also, added some missing quotation marks and other minor fixes.

CxRes committed:
fix Unnecessary Repetition in Response Status

Fixed unnecessary repetition of "Events" in the Subscription Response -> response status.

TallTed commented on protocol.html:
```diff
@@ -451,20 +451,20 @@ <h3 property="schema:name">Overview</h3>
                     <img rel="schema:image" src="notifications-flow-receivefrom.mmd.svg" width="800" />
 
-                    <figcaption property="schema:name">Solid Notifications Flow Receive From, where the receiver establishes connection to sender.</figcaption>
+                    <figcaption property="schema:name">Solid Notifications Flow Receive From, where the a receiver establishes connection to a sender.</figcaption>
```
```suggestion
                    <figcaption property="schema:name">Solid Notifications Flow Receive From, where a receiver establishes connection to a sender.</figcaption>
```

TallTed replied:
or ...
```suggestion
                    <figcaption property="schema:name">Solid Notifications Flow Receive From, where the receiver establishes connection to a sender.</figcaption>
```

TallTed commented on protocol.html:
```diff
@@ -451,20 +451,20 @@ <h3 property="schema:name">Overview</h3>
                     <img rel="schema:image" src="notifications-flow-sendto.mmd.svg" width="800" />
 
-                    <figcaption property="schema:name">Solid Notifications Flow Send To, where the sender delivers notifications to receiver.</figcaption>
+                    <figcaption property="schema:name">Solid Notifications Flow Send To, where the sender delivers notifications to a receiver.</figcaption>
```
should probably match line 459, `a` or `the`
```suggestion
                    <figcaption property="schema:name">Solid Notifications Flow Send To, where a sender delivers notifications to a receiver.</figcaption>
```

TallTed commented on protocol.html:
```diff
@@ -567,7 +567,7 @@ <h2 property="schema:name">Protocol</h2>
               <p id="json-ld-format">This specification uses JSON-LD [<cite><a class="bibref" href="#bib-json-ld11">JSON-LD11</a></cite>] as the preferred data format, and <code>https://www.w3.org/ns/solid/notifications-context/v1</code> as a URI for the JSON-LD context and as a value of the <code>profile</code> parameter used for content negotiation.</p>
 
-              <p id="authentication-authorization">This specification does not require a specific authentication and authorization mechanism to be used with the Solid Notification Protocol. Implementations are encouraged to use existing approaches, such as those described in the Solid Protocol sections on <cite><a href="https://solidproject.org/TR/protocol#authentication" rel="cito:discusses">Aut ...
+              <p id="authentication-authorization">This specification does not require a specific authentication and/or authorization mechanisms to be used with the Solid Notification Protocol. Implementations are encouraged to use existing approaches, such as those described in the Solid Protocol sections on <cite><a href="https://solidproject.org/TR/protocol#authentication" rel="cito:discusses" ...
```
```suggestion
              <p id="authentication-authorization">This specification does not require a specific authentication and/or authorization mechanism to be used with the Solid Notification Protocol. Implementations are encouraged to use existing approaches, such as those described in the Solid Protocol sections on <cite><a href="https://solidproject.org/TR/protocol#authentication" rel="cito:discusses">Authentication</a></cite> and <cite><a href="https://solidproject.org/TR/protocol#authorization" rel="cito:discusses">Authorization</a></cite> [<cite><a class="bibref" href="#bib-solid-protocol">SOLID-PROTOCOL</a></cite>].</p>
```

TallTed commented on quick-protocol.html:
```diff
@@ -463,20 +463,20 @@ <h3 property="schema:name">Overview</h3>
                     <img rel="schema:image" src="notifications-flow-receivefrom.mmd.svg" width="800" />
 
-                    <figcaption property="schema:name">Solid Quick Notifications Flow Receive From, where the receiver establishes connection to sender.</figcaption>
+                    <figcaption property="schema:name">Solid Quick Notifications Flow Receive From, where the receiver establishes a connection to a sender.</figcaption>
```
Again, should match earlier. I think `a` is better than `the`, throughout.
```suggestion
                    <figcaption property="schema:name">Solid Quick Notifications Flow Receive From, where a receiver establishes a connection to a sender.</figcaption>
```

TallTed commented on quick-protocol.html:
```diff
@@ -463,20 +463,20 @@ <h3 property="schema:name">Overview</h3>
                     <img rel="schema:image" src="notifications-flow-sendto.mmd.svg" width="800" />
 
-                    <figcaption property="schema:name">Solid Quick Notifications Flow Send To, where the sender delivers notifications to receiver.</figcaption>
+                    <figcaption property="schema:name">Solid Quick Notifications Flow Send To, where the sender delivers notifications to a receiver.</figcaption>
```
Again...
```suggestion
                    <figcaption property="schema:name">Solid Quick Notifications Flow Send To, where a sender delivers notifications to a receiver.</figcaption>
```

CxRes commented:
@TallTed Thanks again! These batch of changes I was (mostly) aware of, but did not make them because they apply to the main protocol itself! 

Having said that, I will accept them and backport them to SNP (sometime, next week, though).

elf-pavlik commented:
Given PREP and PREP-Solid, do you still plan to pursuit this proposal?

CxRes commented:
Why not? The basic concept is still good...

CxRes commented:
Thank you especially @TallTed, but also @csarven and @elf-pavlik for reviewing this proposal. I am closing this in favour of #201 which is a much simpler and more compatible proposal.
